test/test_puma_server.rb

```diff
@@ -104,7 +104,64 @@ def test_respect_x_forwarded_proto
 
     req = Net::HTTP::Get.new("/")
     req['HOST'] = "example.com"
-    req['X_FORWARDED_PROTO'] = "https"
+    req['X_FORWARDED_PROTO'] = "https,http"
+
+    res = Net::HTTP.start @host, @server.connected_port do |http|
+      http.request(req)
+    end
+
+    assert_equal "443", res.body
+  end
+
+  def test_respect_x_forwarded_ssl_on
+    @server.app = proc do |env|
+      [200, {}, [env['SERVER_PORT']]]
+    end
+
+    @server.add_tcp_listener @host, @port
+    @server.run
+
+    req = Net::HTTP::Get.new("/")
+    req['HOST'] = "example.com"
+    req['X_FORWARDED_SSL'] = "on"
+
+    res = Net::HTTP.start @host, @server.connected_port do |http|
```

Comment:
Hmmm. This doesn't match our current style of just writing onto a TCP socket. Thoughts on that?

Reply:
Eh, I only repeated test bodies which were there, tests that checked similar things, to have something to start with.

When I refactor this, I'll try to narrow down the tests to be more about the methods I changed.